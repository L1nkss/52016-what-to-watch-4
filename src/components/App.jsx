import Main from 'main';
const App = (props) => {
  return (
    <Main filmData={props.data} />
  );
};


App.propTypes = {
  data: PropTypes.arrayOf(
      PropTypes.string
  )
};

export default App;
