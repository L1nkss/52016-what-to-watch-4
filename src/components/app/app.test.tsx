import App from './app';
import * as ShallowRenderer from 'react-test-renderer/shallow';

// Моки
const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideoLink: `img/bohemian-rhapsody.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Drama`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    released: 2015,
    id: 1
  },
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideoLink: `img/bohemian-rhapsody.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Drama`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    released: 2015,
    id: 2
  },
];


describe(`Testing App component`, () => {
  it(`Component should successfully rendered`, () => {
    const myShallowRenderer = ShallowRenderer.createRenderer();
    myShallowRenderer.render(<App
      films={films}
      isDataLoading={false}
      isError={false}
      promoFilm={films[0]}
      loadFilms={() => {}}
      loadPromoFilm={() => {}}
      loadFavoritesFilms={() => {}}
      checkAuthStatus={() => {}}
    />);
    const result = myShallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
