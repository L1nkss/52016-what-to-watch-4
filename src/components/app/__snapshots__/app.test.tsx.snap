// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component Component should successfully rendered 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <Memo(Connect(Main)) />
    </Route>
    <Route
      component={[Function]}
      exact={true}
      path="/film-detail/:id?"
    />
    <Memo(Connect(PrivateRoute))
      exact={true}
      path="/add-review/:id?"
      render={[Function]}
    />
    <Memo(Connect(PrivateRoute))
      exact={true}
      path="/my-list"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/sign-in"
    >
      <Memo(Connect(SignIn)) />
    </Route>
    <Route
      component={[Function]}
      exact={true}
      path="/player/:id?"
    />
    <Route
      exact={true}
      path="/error"
    >
      <Unknown />
    </Route>
    <Route>
      <Unknown />
    </Route>
  </Switch>
</Router>
`;
